pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -0,0 +1,263 @@
+		configMapListerForTargetNamespace: kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Lister().ConfigMaps(targetNamespace),
+		targetNamespace:                   targetNamespace,
+		operandName:                       staticPodName,
+	}
+	return factory.New().
+		ResyncEvery(time.Minute).
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("MissingStaticPodController", eventRecorder)
+}
+
+func (c *missingStaticPodController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	_, originalOperatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	installerPods, err := c.podLister.List(labels.SelectorFromSet(labels.Set{"app": "installer"}))
+	if err != nil {
+		return err
+	}
+
+	// get the most recent installer pod for each node
+	latestInstallerPodsByNode, err := getMostRecentInstallerPodByNode(installerPods)
+	if err != nil {
+		return err
+	}
+
+	errors := make([]string, 0, len(latestInstallerPodsByNode))
+	for node, latestInstallerPodOnNode := range latestInstallerPodsByNode {
+		installerPodRevision, err := getInstallerPodRevision(latestInstallerPodOnNode)
+		if err != nil {
+			// we expect every installer pod to have a installerPodRevision in its name, unexpected error here
+			return fmt.Errorf("failed to get installerPodRevision for installer pod %q - %w", latestInstallerPodOnNode.Name, err)
+		}
+
+		finishedAt, ok := installerPodFinishedAt(latestInstallerPodOnNode)
+		if !ok {
+			// either it's in the process of running, or it ran into an error
+			continue
+		}
+
+		staticPodRevisionOnThisNode := getStaticPodCurrentRevision(node, originalOperatorStatus)
+		gracePeriod, err := c.getStaticPodTerminationGracePeriodSecondsForRevision(staticPodRevisionOnThisNode)
+		if err != nil {
+			return err
+		}
+		threshold := time.Duration(gracePeriod) + 10*time.Second
+
+		if time.Since(finishedAt) > threshold &&
+			staticPodRevisionOnThisNode < installerPodRevision {
+			// if we are here:
+			//  a: the latest installer pod successfully completed at finishedAt
+			//  b: it has been more than 'terminationGracePeriodSeconds' + 10s since the
+			//     installer pod has completed at finishedAt
+			//  c. the static pod is not at correct installerPodRevision yet
+			// then all of the above conditions are true, so we should go degraded
+			syncCtx.Recorder().Eventf("MissingStaticPod", "static pod lifecycle failure - installer pod: %q with revision: %d completed at: %s, static pod revision: %d",
+				latestInstallerPodOnNode.Name, installerPodRevision, finishedAt, staticPodRevisionOnThisNode)
+
+			errors = append(errors, fmt.Sprintf("static pod lifecycle failure - installer pod: %q with revision: %d completed at: %s, static pod revision: %d",
```

Comment:
do we want to report the `threshold` value? (same for the event)

Reply:
oh and maybe, slightly change the msg so that we know why it failed i.e. `static pod for revision: %d didn't show up, waited: %v`